source/unit.js

```diff
@@ -583,7 +588,107 @@ export class Unit {
+   * The calculation is done in two steps:
+   * 1. Calculate the equivalent mass of the substance by dividing its molecular weight by its valence.
+   *    For example, Ca2+ has a valence of 2 and a molecular weight of 40.08 g/mole, so its equivalent mass is 40.08/2 = 20.04 g/equiv.
+   * 2. Calculate the equivalents by dividing the mass by the equivalent mass.
+   * 
+   * "The equivalent weight of an element is its gram atomic weight divided by its valence (combining power)."
+   * https://www.britannica.com/science/equivalent-weight
+   *
+   * @param {number} mass - The mass to be converted.
+   * @param {object} eqUnit - The target/to unit for which the converted number is wanted.
+   * @param {number} molecularWeight - The molecular weight of the substance for which the conversion is being made.
+   * @param {number} valence - The valence of the substance for which the conversion is being made.
+   * @returns {number} - The equivalent amount in the specified equivalent unit.
+   */
+  convertMassToEq(mass, eqUnit, molecularWeight, valence) {
+    // Calculate equivalent mass by dividing molecular weight by valence
+    let equivalentMass = molecularWeight / valence;
+    // Calculate equivalents by dividing mass by equivalent mass
+    let equivalents = mass / equivalentMass;
+    // Get Avogadro's number from the unit tables
+    let avogadroNumber =  this._getUnitTables().getUnitByCode('mol').magnitude_ ;
+    // Calculate mole factor by dividing the magnitude of the equivalent unit by Avogadro's number
+    // eqUnit may have a prefix (e.g. meq) and we need to adjust for that
+    let moleFactor = eqUnit.magnitude_ / avogadroNumber ;
+    // Adjust equivalents by dividing by the mole factor
+    let adjustedEquivalents = equivalents / moleFactor;
+    // Return the adjusted equivalents
+    return adjustedEquivalents;
+  }
+
+  /**
+   * This function converts equivalent amount to moles.
+   * 
+   * @param {number} eqFromVal - The equivalent amount for which the conversion is being made.
+   * @param {object} molToUnit - The target/to unit for which the converted number is wanted.
+   * @param {number} valence - The valence of the substance for which the conversion is being made.
+   * @return {number} - The amount in moles.
+   */
+  convertEqToMol(eqFromVal, molToUnit, valence){
+    // Retrieve the unit tables
+    let tabs = this._getUnitTables();
+    // Retrieve the magnitude of a simple mole unit (Avogadro's number)
+    let avoNum = tabs.getUnitByCode('mol').magnitude_ ;
+    // Calculate the factor by which the current mole unit magnitude exceeds the simple mole unit magnitude
+    let molesFactor = this.magnitude_ / avoNum ;
+    // Retrieve the magnitude of the target mole unit
+    let toMolesFactor = molToUnit.magnitude_ / avoNum ;
+    // Calculate and return the amount in moles, taking into account the valence and the mole factors
+    return (eqFromVal * molesFactor ) / valence / toMolesFactor;
```

Comment:
It looks like Avogadro's number above cancels out.  It might introduce rounding errors to have the intermediate steps.  You might keep the intermediate steps as comments, and then do the calculation using a simplified formula.

Reply:
Yeah, since its a relative conversion of mol to mol essentially, turns out I don't need Avogadro's number. Applies to convertMolToEq. This simplifies the function a lot